File: src/main/java/com/vinyl/model/Record.java
package com.vinyl.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Record {
	private String recordBarcode;
	private String releaseBarcodeFk;
	private Integer checkNum;
	private String supplierEdrpou;
	private Date purchaseDate;
	private Integer purchasePrice;
	private Integer sellPrice;
	private Boolean available;
	private String recordState;
	private Date stateCheckDate;
}

File: src/main/resources/frontend/src/schemas/record.js
export default {
  props: {
    recordBarcode: 'string',
    releaseBarcodeFk: {
      type: 'release',
      isSchema: true
    },
    purchaseDate: {
      type: 'date'
    },
    purchasePrice: 'int',
    sellPrice: 'int',
    available: true,
    recordState: 'M|NM|VG+|VG|G|F',
    stateCheckDate: 'date'
  },
  key: 'recordBarcode',
  display: (item) => `${item.releaseBarcodeFk} - ${item.recordState}`
}